py/desitarget/mock/mockmaker.py

```diff
@@ -2242,7 +2244,11 @@ def readmock(self, mockfile=None, healpixels=None, nside=None,
         # Build the full filenames.
         lyafiles = []
         for mpix in mockpix:
-            lyafiles.append("%s/%d/%d/transmission-%d-%d.fits"%(
+            try: #recent fits extension
+                lyafiles.append("%s/%d/%d/transmission-%d-%d.fits.gz"%(
+                mockdir, mpix//100, mpix, nside_lya, mpix))
+            except: # old fits extension
+                lyafiles.append("%s/%d/%d/transmission-%d-%d.fits"%(
                 mockdir, mpix//100, mpix, nside_lya, mpix))
 
```

Comment:
Here you're just constructing a string, so I don't see how the `try` portion of the code block would ever fail. Or am I missing something?

Reply:
Hi, I used the try statement because only the latest versions of the mocks have the .gz extension. It will crash for older versions otherwise. If a IF statement is better I can change it.

Reply:
Hi @moustakas, I've fixed this lines. And just added an IF statement in lines L3372-3379  to check if the file exist, if not it will try with the old extension, and raise an error otherwise.